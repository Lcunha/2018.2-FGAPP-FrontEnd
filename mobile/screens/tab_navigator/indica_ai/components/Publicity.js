import React, { Component}   from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  ImageBackground
} from 'react-native';
import { Container, Header, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body, Right, Row } from 'native-base';
import { withNavigation } from 'react-navigation';
class Publicity extends Component {
  constructor(props) {
    super(props);
    this.state = {
      onPress: props.onPress,
    }
  }

  render() {

    const {name, address , local_images} = this.props.local
    const image  = (local_images.length !== 0) ?
      {uri: "data:image/jpg;base64," + local_images[local_images.length - 1]["image"]}
      :require('../assets/IntegraApps_icon.png')

    return (
      <TouchableOpacity onPress={() => {
        this.props.navigation.navigate('LocalDetails',{
          local: this.props.local
        });
      }}>
        <Card>
          <CardItem cardBody style={{paddingHorizontal: 5, paddingTop: 5}}>
            <ImageBackground source={image} style={{height: 200, width: null, flex: 1}}>
            </ImageBackground>
          </CardItem>
          <CardItem>
            <Left>
              <Thumbnail source={require('../assets/fga-logo.png')} />
              <Body>
                <Text style={styles.localName}>{name}</Text>
                <Text note>{address}</Text>
              </Body>
            </Left>
          </CardItem>
          <CardItem style={{justifyContent: 'flex-end'}}>
            <Text style={styles.localAnuncio}>An√∫ncio</Text>
          </CardItem>
        </Card>
      </TouchableOpacity>
    );
  }
}

export default withNavigation(Publicity);
const styles = StyleSheet.create({

  localAnuncio: {
    color: '#0AACCC',
    fontSize: 13,
    borderWidth: 1,
    borderColor: '#0AACCC',
    borderRadius: 2
  },
  localName: {
    fontWeight: 'bold',
    fontSize: 20
  }

});
