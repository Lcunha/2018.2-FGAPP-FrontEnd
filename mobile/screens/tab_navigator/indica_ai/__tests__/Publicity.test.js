import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

import Publicity from '../components/Publicity';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<Publicity />).toJSON();
    expect(tree).toMatchSnapshot();
});
