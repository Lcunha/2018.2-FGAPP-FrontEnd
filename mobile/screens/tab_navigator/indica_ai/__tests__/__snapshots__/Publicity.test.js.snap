// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
    virtual={undefined}
  >
    <View
      cardBody={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": -5,
            "paddingLeft": -5,
            "paddingRight": -5,
            "paddingTop": -5,
          },
          Object {
            "paddingHorizontal": 5,
            "paddingTop": 5,
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 200,
            "width": null,
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 200,
                "width": null,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            }
          }
          virtual={undefined}
        />
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": "center",
              "flex": 1,
              "marginLeft": 10,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            note={true}
            style={
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "marginRight": 20,
              }
            }
            uppercase={false}
            virtual={undefined}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 10,
                },
                Object {
                  "color": "#0AACCC",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 0.8,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": null,
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 15,
              },
              Object {
                "borderColor": "#0AACCC",
                "borderRadius": 2,
                "borderWidth": 1,
                "color": "#0AACCC",
                "fontSize": 13,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          An√∫ncio
        </Text>
      </View>
    </View>
  </View>
</View>
`;
